bee-storage/src/sqlx/test.rs

```diff
@@ -0,0 +1,388 @@
+extern crate bee_bundle;
+extern crate rand;
+
+#[cfg(test)]
+mod tests {
+
+    const BEE_TEST_DB_USER: &str = "test_db_user";
+    const BEE_TEST_DB_NAME: &str = "test_db";
+
+    use crate::{
+        sqlx::SqlxBackendStorage,
+        storage::{
+            HashesToApprovers,
+            MissingHashesToRCApprovers,
+            StateDeltaMap,
+            StorageBackend,
+        },
+    };
+
+    use futures::{
+        executor::block_on,
+        future::join_all,
+    };
+
+    use std::{
+        collections::{
+            HashMap,
+            HashSet,
+        },
+        io::{
+            self,
+            Write,
+        },
+        panic,
+        process::Command,
+        rc::Rc,
+        time::Instant,
+    };
+
+    use bee_bundle::{
+        Address,
+        Hash,
+        Payload,
+        TransactionField,
+    };
+
+    fn test_db_url() -> String {
+        format!(
+            "postgres://{}:dummy_password@localhost/{}",
+            BEE_TEST_DB_USER, BEE_TEST_DB_NAME
+        )
+    }
+
+    fn run_test<T>(test: T) -> ()
+        where
+            T: FnOnce() -> () + panic::UnwindSafe,
+    {
+        setup_db();
+
+        let result = panic::catch_unwind(|| test());
+
+        teardown_db();
+
+        assert!(result.is_ok())
+    }
+
+    fn setup_db() -> () {
```

Comment:
Can we try to find a way to make these tests accessible by all the backends ? All the tests should be able to operate on a storage trait, right ? Only the init/shutdown may differ. It would probably help you when working on RocksDB.

Reply:
I guess we can move `run_test` to `bee_test` so it can bee later be used by `bee_tangle` and `bee_consensus`?
and add  `trait Testable { fn setup(); fn teardown();}` ?
can do that on the rocksdb PR

basically:
```
    fn run_test<T, U >(test: T ) -> ()
        where
            T: FnOnce() -> () + panic::UnwindSafe, 
            U: Testable
{
        setup();

        let result = panic::catch_unwind(|| test());

        teardown();

        assert!(result.is_ok())
}
```
though we could figure out better names and that would have to wait to the next PR

Reply:
Sure that can wait. Just want to be sure we're not going to rewrite these tests for RocksDB or whatever.

Reply:
It might even make sense to add `trait StorageTest` with the test suite as methods in addition